### ISSUE TITLE
problem reading geotiff with terra::rast

### ISSUE THREAD

**Issue by hansvancalster** [rspatial/terra#282]
The example below illustrates some unexpected behavior after reading a `.tif` file with `terra::rast()`. 
Reading the same file with `raster::raster()` does not have these problems.
It looks like `terra::rast()` reads the values as a factor named Count with levels = values and labels = counts (the number of cells with that particular value)?
I'm not sure if this is a problem with the specific `.tif` file (which was exported from an ArcGis gdb), or something that should be treated differently by `terra` (given that `raster::raster()` seems to behave differently).

``` r
tiffile <- "Z:/Projects/PRJ_NARA_2020/Basisbestanden/landgebruik_geotiff/LG2016_finaal_natuurbeheer.tif"
tr <- terra::rast(tiffile)
rr <- raster::raster(tiffile)
#> Warning in showSRID(SRS_string, format = "PROJ", multiline = "NO", prefer_proj =
#> prefer_proj): Discarded datum Reseau National Belge 1972 in Proj4 definition

# raster::raster prints correctly and shows values (1 to 30)
rr
#> class      : RasterLayer 
#> dimensions : 13373, 25685, 343485505  (nrow, ncol, ncell)
#> resolution : 10, 10  (x, y)
#> extent     : 2150, 259000, 153000, 286730  (xmin, xmax, ymin, ymax)
#> crs        : +proj=lcc +lat_0=90 +lon_0=4.36748666666667 +lat_1=51.1666672333333 +lat_2=49.8333339 +x_0=150000.013 +y_0=5400088.438 +ellps=intl +units=m +no_defs 
#> source     : LG2016_finaal_natuurbeheer.tif 
#> names      : LG2016_finaal_natuurbeheer 
#> values     : 1, 30  (min, max)

# terra::rast prints incorrect and shows counts
tr
#> class       : SpatRaster 
#> dimensions  : 13373, 25685, 1  (nrow, ncol, nlyr)
#> resolution  : 10, 10  (x, y)
#> extent      : 2150, 259000, 153000, 286730  (xmin, xmax, ymin, ymax)
#> coord. ref. : +proj=lcc +lat_0=90 +lon_0=4.36748666666667 +lat_1=51.1666672333333 +lat_2=49.8333339 +x_0=150000.013 +y_0=5400088.438 +ellps=intl +units=m +no_defs 
#> source      : LG2016_finaal_natuurbeheer.tif 
#> name        :    Count 
#> min value   :     8505 
#> max value   : 45752781

# a plot of terra object is incorrect and shows counts instead of values
terra::plot(tr)
```

![](https://i.imgur.com/924f2oV.png)

``` r
# extracting values with dataframe = FALSE returns correct values
terra::values(tr, row = 10000, nrows = 5, col = 5000, ncols = 5, 
              dataframe = FALSE)
#>       Count
#>  [1,]    11
#>  [2,]    11
#>  [3,]    11
#>  [4,]    11
#>  [5,]    11
#>  [6,]    11
#>  [7,]    11
#>  [8,]    11
#>  [9,]    11
#> [10,]    11
#> [11,]    11
#> [12,]    11
#> [13,]    11
#> [14,]    11
#> [15,]    11
#> [16,]    11
#> [17,]    11
#> [18,]    11
#> [19,]    11
#> [20,]    11
#> [21,]    11
#> [22,]    11
#> [23,]    11
#> [24,]    11
#> [25,]    11

# extracting values with dataframe = TRUE returns counts instead of values
terra::values(tr, row = 10000, nrows = 5, col = 5000, ncols = 5, 
              dataframe = TRUE)
#>      Count
#> 1  2322978
#> 2  2322978
#> 3  2322978
#> 4  2322978
#> 5  2322978
#> 6  2322978
#> 7  2322978
#> 8  2322978
#> 9  2322978
#> 10 2322978
#> 11 2322978
#> 12 2322978
#> 13 2322978
#> 14 2322978
#> 15 2322978
#> 16 2322978
#> 17 2322978
#> 18 2322978
#> 19 2322978
#> 20 2322978
#> 21 2322978
#> 22 2322978
#> 23 2322978
#> 24 2322978
#> 25 2322978
```

<sup>Created on 2021-07-15 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> - Session info ---------------------------------------------------------------
#>  setting  value                       
#>  version  R version 4.1.0 (2021-05-18)
#>  os       Windows 10 x64              
#>  system   x86_64, mingw32             
#>  ui       RTerm                       
#>  language (EN)                        
#>  collate  Dutch_Belgium.1252          
#>  ctype    Dutch_Belgium.1252          
#>  tz       Europe/Paris                
#>  date     2021-07-15                  
#> 
#> - Packages -------------------------------------------------------------------
#>  package     * version date       lib source        
#>  cli           2.5.0   2021-04-26 [1] CRAN (R 4.0.5)
#>  codetools     0.2-18  2020-11-04 [2] CRAN (R 4.1.0)
#>  curl          4.3.1   2021-04-30 [1] CRAN (R 4.1.0)
#>  digest        0.6.27  2020-10-24 [1] CRAN (R 4.1.0)
#>  evaluate      0.14    2019-05-28 [1] CRAN (R 4.1.0)
#>  fs            1.5.0   2020-07-31 [1] CRAN (R 4.1.0)
#>  glue          1.4.2   2020-08-27 [1] CRAN (R 4.1.0)
#>  highr         0.9     2021-04-16 [1] CRAN (R 4.1.0)
#>  htmltools     0.5.1.1 2021-01-22 [1] CRAN (R 4.1.0)
#>  httr          1.4.2   2020-07-20 [1] CRAN (R 4.1.0)
#>  knitr         1.33    2021-04-24 [1] CRAN (R 4.1.0)
#>  lattice       0.20-44 2021-05-02 [2] CRAN (R 4.1.0)
#>  magrittr      2.0.1   2020-11-17 [1] CRAN (R 4.1.0)
#>  mime          0.11    2021-06-23 [1] CRAN (R 4.0.5)
#>  ps            1.6.0   2021-02-28 [1] CRAN (R 4.1.0)
#>  R6            2.5.0   2020-10-28 [1] CRAN (R 4.1.0)
#>  raster        3.4-13  2021-06-18 [1] CRAN (R 4.0.5)
#>  Rcpp          1.0.6   2021-01-15 [1] CRAN (R 4.1.0)
#>  reprex        2.0.0   2021-04-02 [1] CRAN (R 4.1.0)
#>  rgdal         1.5-23  2021-02-03 [1] CRAN (R 4.1.0)
#>  rlang         0.4.11  2021-04-30 [1] CRAN (R 4.1.0)
#>  rmarkdown     2.9     2021-06-15 [1] CRAN (R 4.0.5)
#>  rstudioapi    0.13    2020-11-12 [1] CRAN (R 4.1.0)
#>  sessioninfo   1.1.1   2018-11-05 [1] CRAN (R 4.1.0)
#>  sp            1.4-5   2021-01-10 [1] CRAN (R 4.1.0)
#>  stringi       1.6.2   2021-05-17 [1] CRAN (R 4.0.5)
#>  stringr       1.4.0   2019-02-10 [1] CRAN (R 4.1.0)
#>  terra         1.2-10  2021-05-13 [1] CRAN (R 4.1.0)
#>  withr         2.4.2   2021-04-18 [1] CRAN (R 4.1.0)
#>  xfun          0.24    2021-06-15 [1] CRAN (R 4.0.5)
#>  xml2          1.3.2   2020-04-23 [1] CRAN (R 4.1.0)
#>  yaml          2.2.1   2020-02-01 [1] CRAN (R 4.1.0)
#> 
#> [1] C:/R/library
#> [2] C:/R/R-4.1.0/library
```

</details>

**Comment by tungmilan** [comment]
Have you tried the latest development version yet?
```
Version: 1.3-15
Date: 2021-07-12
```

**Comment by rhijmans** [comment]
Indeed it looks like the file has an attribute table; but one that only has "Counts" . In this case, the value `11` occurs `2322978` times (according to the Raster Attribute Table).  I try to ignore these degenerate attributes (that only have Counts), but I may have missed this case. 

It would be useful to know what version of `terra` you are using. And if you are using the current CRAN version, it would also be helpful if you tried it with the development version. It is easy to install with 
`install.packages('terra', repos='https://rspatial.r-universe.dev')`

**Comment by hansvancalster** [comment]
I was using an older CRAN version (1.2-10). Sorry about that. After updating to the latest version on CRAN (1.3-4), everything worked as expected. Probably due to the fix provided in #234. For completeness, I also checked the latest dev version and (as expected) everything worked.
Thank you both for your advice and for a very much appreciated R package 👍

**Comment by Rapsodia86** [comment]
The problem came back. I checked the #234 but again, it does not read pixel values properly.
I used extract by mask (shapefile) to clip the original NLCD *.img in ArcMap and saved as *.tif. Then when read in R, using raster::raster(), the clipped *.tif is ok. Meaning, the values are numerical (not categorical anymore), but they are correct.
However, terra::rast() reads counts as values like in the example above. 

I am using 1.4-13  dev version (#install.packages('terra', repos='https://rspatial.r-universe.dev')).
